simpleclient_httpserver/src/main/java/io/prometheus/client/exporter/HTTPServer.java

```diff
@@ -54,16 +60,21 @@ protected ByteArrayOutputStream initialValue()
         public void handle(HttpExchange t) throws IOException {
             String query = t.getRequestURI().getRawQuery();
 
+            String contextPath = t.getHttpContext().getPath();
             ByteArrayOutputStream response = this.response.get();
```

Comment:
Not related to this PR, but why do we cache the output stream?

Reply:
I believe it because, we dont want to create new object. Anyways we are writing metrics to  bytestream for each call.